### PR TITLE
docs(list): add component playgrounds

### PR DESCRIPTION
**Pull request by brandyscarney**
Adds the following playgrounds:

- Basic Usage
- Inset List (I added a different content background here but didn't put it in the source code for the demos, can add if needed)
- List Lines

I considered adding an example of a reorder group in the list, but linked to it instead. 

https://ionic-docs-git-fw-1284-ionic1.vercel.app/docs/api/list

### CODE REVIEW

File: static/usage/list/basic/react.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          <IonLabel>The Legend of Zelda</IonLabel>
+        </IonItem>
+        <IonItem>
+          <IonLabel>Pac-Man</IonLabel>
+        </IonItem>
+        <IonItem>
+          <IonLabel>Super Mario World</IonLabel>
+        </IonItem>
+      </IonList>
+    </>
```
Comment (amandaejohnston) [inline static/usage/list/basic/react.md:25]:
```suggestion
    <IonList>
      <IonItem>
        <IonLabel>Pokémon Yellow</IonLabel>
      </IonItem>
      <IonItem>
        <IonLabel>Mega Man X</IonLabel>
      </IonItem>
      <IonItem>
        <IonLabel>The Legend of Zelda</IonLabel>
      </IonItem>
      <IonItem>
        <IonLabel>Pac-Man</IonLabel>
      </IonItem>
      <IonItem>
        <IonLabel>Super Mario World</IonLabel>
      </IonItem>
    </IonList>
```

File: docs/api/list.md [hunk 2]
```diff
@@ -32,411 +25,36 @@ import EncapsulationPill from '@components/page/api/EncapsulationPill';
...
-      <ion-label>The Legend of Zelda</ion-label>
-    </ion-item>
-    <ion-item>
-      <ion-label>Pac-Man</ion-label>
-    </ion-item>
-    <ion-item>
-      <ion-label>Super Mario World</ion-label>
-    </ion-item>
-  </ion-list>
+<Inset />
```
Comment (amandaejohnston) [inline docs/api/list.md:46]: `md` mode shows an extra line below the first item here 🤔 Is this an Ionic bug?
extra line

File: docs/api/list.md [hunk 3]
```diff
@@ -32,411 +25,36 @@ import EncapsulationPill from '@components/page/api/EncapsulationPill';
...
-      <ion-item>
-        <ion-label>Item</ion-label>
-      </ion-item>
-      <ion-item-options side="end">
-        <ion-item-option @click="unread(item)">Unread</ion-item-option>
-      </ion-item-options>
-    </ion-item-sliding>
-  </ion-list>
-</template>
+Adding the `lines` property to a list will adjust the bottom borders of all of the items in the list. Setting it to `"full"` will display full width borders, `"inset"` will display borders adjusted with left padding, and `"none"` will show no borders. If the `lines` property is set on an item in a list that will take priority over the property on the list.
```
Comment (amandaejohnston) [inline docs/api/list.md:51]:
```suggestion
Adding the `lines` property to a list will adjust the bottom borders of all of the items in the list. Setting it to `"full"` will display full width borders, `"inset"` will display borders adjusted with left padding, and `"none"` will show no borders. If the `lines` property is set on an item in a list, that will take priority over the property on the list.
```

**Review by amandaejohnston — APPROVED** [review]
LGTM aside from these comments 👍

File: static/usage/list/inset/angular.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+```html
+<ion-list inset>
```
Comment (liamdebeasi) [inline static/usage/list/inset/angular.md:2]:
```suggestion
<ion-list [inset]="true">
```

File: static/usage/list/inset/javascript.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+```html
+<ion-list inset>
```
Comment (liamdebeasi) [inline static/usage/list/inset/javascript.md:2]:
```suggestion
<ion-list inset="true">
```

File: static/usage/list/inset/react.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+```tsx
+import React from 'react';
+import { IonItem, IonLabel, IonList } from '@ionic/react';
+
+function Example() {
+  return (
+    <IonList inset>
```
Comment (liamdebeasi) [inline static/usage/list/inset/react.md:7]:
```suggestion
    <IonList inset={true}>
```

File: static/usage/list/inset/vue.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+```html
+<template>
+  <ion-list inset>
```
Comment (liamdebeasi) [inline static/usage/list/inset/vue.md:3]:
```suggestion
  <ion-list :inset="true">
```

**Review by liamdebeasi — APPROVED** [review]
Good to go when the below comments are addressed.

**Pull request merged**